lib/Optimizer/Transforms/Utils.cpp

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "cudaq/Optimizer/Dialect/CC/CCOps.h"
+
+#include <complex>
+#include <vector>
+
+using namespace mlir;
+
+namespace cudaq {
+
+std::vector<std::complex<double>>
+readGlobalConstantArray(cudaq::cc::GlobalOp &global) {
```

Comment:
```suggestion
cudaq::factory::readGlobalConstantArray(cudaq::cc::GlobalOp &global) {
```
(see LLVM style guide)

Reply:
Removed this file, moved to `Factory.cpp` in 3e71dee865f261ba1ba9379e172eb4b68ccab2d7